### PR TITLE
Upgrade to Node 20

### PR DESCRIPTION
**Pull request by jbudz**
Depends on https://github.com/elastic/ci-agent-images/pull/429

We're looking into upgrading Kibana to Node 20 but running into bootstrap issues:
```
[bazel] error @elastic/ems-client@8.5.0: The engine "node" is incompatible with this module. Expected version ">=16<=18". Got "20.5.0"
[bazel] error Found incompatible module.
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,6 +72,6 @@
     "typescript": "4.6.3"
   },
   "engines": {
-    "node": ">=16 <=18"
+    "node": ">=16 <=20"
```
Comment (jbudz) [inline package.json:75]: Now that Node 16 is EOL?
```suggestion
    "node": ">=18 <=20"
```

**Comment by jsanz** [comment]
Buildkite test this

**Review by jsanz — APPROVED** [review]
Changes are OK to me, I'll also go ahead and merge this change so I can publish an update of the library in NPM today and open a PR in Kibana to fix this issue.

cc. @nickpeihl

**Comment by jsanz** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.4|[](https://github.com/elastic/ems-client/pull/210)|
|✅|7.17|[](https://github.com/elastic/ems-client/pull/211)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**